### PR TITLE
Support enum strings

### PR DESCRIPTION
**Pull request by LukasJenicek**
Proposal: https://github.com/webrpc/webrpc/issues/203

### CODE REVIEW

File: _examples/golang-basics/example.gen.go [hunk 1]
```diff
@@ -78,13 +78,44 @@ func (x *Kind) Is(values ...Kind) bool {
...
 }
 
 type User struct {
-	ID       uint64 `json:"id" db:"id"`
-	Username string `json:"USERNAME" db:"username"`
-	Role     string `json:"role" db:"-"`
+	ID       uint64  `json:"id" db:"id"`
+	Username string  `json:"USERNAME" db:"username"`
+	Role     string  `json:"role" db:"-"`
+	Kind     *Kind   `json:"kind"`
```
Comment (pkieltyka) [inline _examples/golang-basics/example.gen.go:117]: perhaps its for historical reasons, but do you guys remember why we default to using a `*` for the enum values..? as these are just integers or strings, I feel we should drop the pointer... thoughts..?

**Review by pkieltyka — COMMENTED** [review]
very nice work on this -- thanks for the welcome addition.

I left one comment

File: _examples/golang-basics/example.gen.go (hunk 1 above)
Comment (pkieltyka) [reply _examples/golang-basics/example.gen.go:117]: FYI, if someone defines a field as, `kind?: Kind` then yes it should output *Kind, but if the definition is just `kind: Kind` then I believe it should just be of type `Kind`

Comment (pkieltyka) [reply _examples/golang-basics/example.gen.go:117]: see also reported in the past https://github.com/webrpc/webrpc/issues/55

Comment (VojtechVitek) [reply _examples/golang-basics/example.gen.go:117]: I don't see a reason for this. Agreed.

Comment (LukasJenicek) [reply _examples/golang-basics/example.gen.go:133]: It's a problem on `gen-golang` template side see: https://github.com/webrpc/gen-golang/pull/53

**Comment by VojtechVitek** [comment]
LGTM

**Pull request merged**